drop incorrect index on user_messages_edits and create index on user_messages_deletes

michaelsbradleyjr opened the pull request:
In `protocol/migrations/sqlite/1625762506_add_deleted_messages.up.sql` an index is created on `user_messages_edits` but it should instead be created on `user_messages_deletes`.

Drop the incorrect index and create the correct one.

---

I installed `go-bindata` via `make gen-install`, and I ran `go generate` within the `protocol/migrations/sqlite` directory.

Let me know if I missed a step or did something incorrectly and I'll fix it ASAP.

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status_go.html))?
- [ ] Have you tested changes with mobile or desktop version?

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (3)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 563309ff | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2358/1/) | 2021-09-08 20:01:21 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210908-195857-563309.zip) |
| :heavy_check_mark: | 563309ff | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2358/1/) | 2021-09-08 20:02:07 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210908-195852-563309.zip) |
| :heavy_check_mark: | 563309ff | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2358/1/) | 2021-09-08 20:05:40 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210908-195852-563309.aar) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | cafb9832 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2358/2/) | 2021-10-04 06:22:16 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-211004-061838-cafb98-pr2358.zip) |
| :heavy_check_mark: | cafb9832 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2358/2/) | 2021-10-04 06:22:44 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-211004-061838-cafb98-pr2358.zip) |
| :heavy_check_mark: | cafb9832 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2358/2/) | 2021-10-04 06:25:05 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-211004-061838-cafb98-pr2358.aar) |
| | | | | | | |
| :heavy_check_mark: | 006150c8 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2358/3/) | 2021-10-05 18:58:16 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-211005-185604-006150-pr2358.zip) |
| :heavy_check_mark: | 006150c8 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2358/3/) | 2021-10-05 18:58:25 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-211005-185609-006150-pr2358.zip) |
| :heavy_check_mark: | 006150c8 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2358/3/) | 2021-10-05 19:02:10 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-211005-185604-006150-pr2358.aar) |

Samyoul commented:
Hey @michaelsbradleyjr the filename of your migration should have the `*.up.sql` suffix otherwise it won't be picked up by the migrator.

There are also a helpful make commands that will prefix with a unix timestamp and the correct suffixes for you,

- `make migration D=<MIGRATION NAME>`
  - Outputs to `appdatabase/migrations/sql/<MIGRATION FILE>`
- `make migration-protocol D=<MIGRATION NAME>`
  - Outputs to `protocol/migrations/sqlite/<MIGRATION FILE>`

michaelsbradleyjr commented:
Sorry to leave this hanging, I didn't notice the email with @Samyoul's reply and just now thought to check back on this PR.

@Samyoul I'm happy to add the `.up.sql` suffix. When I created the file at the time, I saw there was already a file named [`1628280060_create-usermessages-index.sql`](https://github.com/status-im/status-go/blob/fix/deletes-index/protocol/migrations/sqlite/1628280060_create-usermessages-index.sql) (lacking the suffix) and it was picked up by the migrator, so I named mine similarly since it likewise only deals with an index.

Should I rename both files so they end with `.up.sql` or only mine?

Samyoul commented:
Hey @michaelsbradleyjr that's interesting. Are you able to see that the new indexes are created? I can see that `make generate` picks up the migration file to create a comm, but without the `.up.sql` suffix the migrator can't know if a migration is up or down.

I've checked this migration and the one you mention directly in the db, it isn't being picked up.

<img width="462" alt="Screenshot 2021-09-30 at 16 16 16" src="https://user-images.githubusercontent.com/5702426/135483900-319482a9-b433-4e60-b5eb-e9e6328011e8.png">

<img width="462" alt="Screenshot 2021-09-30 at 16 18 00" src="https://user-images.githubusercontent.com/5702426/135483916-ff63c3d2-0c65-4a4e-9fa4-90b3b1959bf2.png">

michaelsbradleyjr commented:
> Are you able to see that the new indexes are created?

No, actually! Good catch.

I had conflated seeing the incorrect index added in `1625762506_add_deleted_messages.up.sql` (in a command-line dump of the schema) with the index added in `1628280060_create-usermessages-index.sql`.

So I should change the name of `1628280060_create-usermessages-index.sql` to `1628280060_create-usermessages-index.up.sql` and `1631129360_fix-deletedmessages-index.sql` to `1631129360_fix-deletedmessages-index.up.sql`, and that's it?

Samyoul commented:
>So I should change the name of 1628280060_create-usermessages-index.sql to 1628280060_create-usermessages-index.up.sql and 1631129360_fix-deletedmessages-index.sql to 1631129360_fix-deletedmessages-index.up.sql, and that's it?

Yes, that should be enough for the migrator to pick up the new migrations

michaelsbradleyjr commented:
Done!

michaelsbradleyjr committed:
drop incorrect index on user_messages_edits and create index on user_messages_deletes

Also rename `1628280060_create-usermessages-index.sql` to
`1628280060_create-usermessages-index.up.sql`.

churik commented:
@michaelsbradleyjr is PR still relevant?

igor-sirotin commented:
Closing as too old. Feel free to re-open and rebase.
